### DISCUSSION TITLE
opentcs rmi-api web-api

### DISCUSSION THREAD

**Discussion by Tmakerchima** [openTCS/opentcs#112] [Q&A]
I start a springboot framework backup as a client.I got some kernel data via rmi-api.However,everytime after i take a operation like move the vehicle to somewhere.How can i upload this operation to the kernel.I can not find any updateXXX usage in the KernelServicePortal.

**Comment by martingr** [comment]
In openTCS, a vehicle driver is responsible for informing the kernel about updates to a vehicle's state, such as a new position. (This is not done via the RMI API or the web API.) How to implement a vehicle driver is described in the openTCS developer's guide (see https://www.opentcs.org/docs/5.11/developer/developers-guide/opentcs-developers-guide.html#_developing_vehicle_drivers). For example, in order to update a vehicle's position in the kernel, the vehicle driver needs to invoke the `setVehiclePosition()` method on its `VehicleProcessModel`.
